File: scripts/openclaw_adapter/source_mapper.py
from __future__ import annotations


def map_ingestion_path(source_type: str, source_name: str, metadata: dict | None) -> str:
    source_type = (source_type or "").strip().lower()
    source_name = (source_name or "").strip().lower()

    mapping = {
        "whatsapp": "network_responses",
        "telegram": "network_responses",
        "slack": "network_responses",
        "webchat": "network_responses",
        "web_search": "network_responses",
        "web_fetch": "network_responses",
        "external_api": "network_responses",
        "notifications": "notifications",
        "android_notif": "notifications",
        "attachment": "shared_storage",
        "imported_doc": "shared_storage",
        "sync_file": "shared_storage",
        "clipboard": "clipboard",
        "intent": "android_intents",
        "deep_link": "android_intents",
        "accessibility": "ui_accessibility",
        "ocr": "ui_accessibility",
        "screen_ui": "ui_accessibility",
        "ui": "ui_accessibility",
        "memory_chunk": "rag_store",
        "rag": "rag_store",
        "retrieval": "rag_store",
    }

    for candidate in (source_name, source_type):
        if candidate in mapping:
            return mapping[candidate]

    if metadata:
        tool_name = str(metadata.get("tool_name", "")).strip().lower()
        if tool_name in mapping:
            return mapping[tool_name]

    return "network_responses"

File: extensions/openclaw-prism/src/source_mapper.ts
const NETWORK = new Set([
  "whatsapp",
  "telegram",
  "slack",
  "webchat",
  "web_search",
  "web_fetch",
  "external_api",
]);

const NOTIFICATIONS = new Set(["notifications", "android_notif"]);
const STORAGE = new Set(["attachment", "imported_doc", "sync_file"]);
const CLIPBOARD = new Set(["clipboard"]);
const INTENTS = new Set(["intent", "deep_link"]);
const UI = new Set(["accessibility", "ocr", "screen_ui", "ui"]);
const MEMORY = new Set(["memory_chunk", "rag", "retrieval"]);
const TRUSTED = new Set(["system_prompt", "operator_config", "plugin_code", "bootstrap"]);

export function mapSource(event: string, toolName?: string, sourceName?: string): string {
  const candidates = [sourceName, toolName, event]
    .map((value) => (value ?? "").trim().toLowerCase())
    .filter(Boolean);

  for (const candidate of candidates) {
    if (NETWORK.has(candidate)) return "network_responses";
    if (NOTIFICATIONS.has(candidate)) return "notifications";
    if (STORAGE.has(candidate)) return "shared_storage";
    if (CLIPBOARD.has(candidate)) return "clipboard";
    if (INTENTS.has(candidate)) return "android_intents";
    if (UI.has(candidate)) return "ui_accessibility";
    if (MEMORY.has(candidate)) return "rag_store";
  }

  return "network_responses";
}

export function isTrusted(sourceName: string): boolean {
  return TRUSTED.has((sourceName ?? "").trim().toLowerCase());
}

